dot/network/service.go

```diff
@@ -595,6 +600,24 @@ func (s *Service) GetRequestResponseProtocol(subprotocol string, requestTimeout
 	}
 }
 
+func (s *Service) GetNetworkEventsChannel() <-chan NetworkEventInfo {
```

Comment:
is this intended to be only called once? Sharing a channel doesn't guarantee delivery to all consumers of the channel.

Reply:
Right now it is being used by only one consumer. I am not aware of anything other code parts that could become a consumer in future.
Instead of passing around the channel directly, I am using a method, since that way I can call it with `Network` interface

Reply:
addressed. a consumer can now call a method and get a channel